src/order/symm_diff.lean

```diff
@@ -0,0 +1,115 @@
+rfl
+
+lemma symm_diff_eq_xor (p q : Prop) : p Δ q = xor p q := rfl
+
+section generalized_boolean_algebra
+variables {α : Type*} [generalized_boolean_algebra α] (a b c : α)
+
+lemma symm_diff_comm : a Δ b = b Δ a := by simp only [(Δ), sup_comm]
+
+lemma sdiff_symm_diff : c \ (a Δ b) = (c ⊓ a ⊓ b) ⊔ ((c \ a) ⊓ (c \ b)) :=
+by simp only [(Δ), sdiff_sdiff_sup_sdiff']
+
+lemma sdiff_symm_diff' : c \ (a Δ b) = (c \ (a ⊔ b)) ⊔ (c ⊓ a ⊓ b) :=
+by rw [sdiff_symm_diff, sdiff_sup, sup_comm]
+
+lemma symm_diff_sdiff : (a Δ b) \ c = (a \ (b ⊔ c)) ⊔ (b \ (a ⊔ c)) :=
+by rw [symm_diff_def, sup_sdiff, sdiff_sdiff_left, sdiff_sdiff_left]
+
+lemma symm_diff_symm_diff_left :
+  a Δ b Δ c = (a \ (b ⊔ c)) ⊔ (b \ (a ⊔ c)) ⊔ (c \ (a ⊔ b)) ⊔ (a ⊓ b ⊓ c) :=
+calc a Δ b Δ c = ((a Δ b) \ c) ⊔ (c \ (a Δ b))   : symm_diff_def _ _
+           ... = (a \ (b ⊔ c)) ⊔ (b \ (a ⊔ c)) ⊔
+                   ((c \ (a ⊔ b)) ⊔ (c ⊓ a ⊓ b)) : by rw [sdiff_symm_diff', symm_diff_sdiff]
+           ... = (a \ (b ⊔ c)) ⊔ (b \ (a ⊔ c)) ⊔
+                   (c \ (a ⊔ b)) ⊔ (a ⊓ b ⊓ c)   : by ac_refl
+
+lemma symm_diff_symm_diff_right :
+  a Δ (b Δ c) = (a \ (b ⊔ c)) ⊔ (b \ (a ⊔ c)) ⊔ (c \ (a ⊔ b)) ⊔ (a ⊓ b ⊓ c) :=
+calc a Δ (b Δ c) = (a \ (b Δ c)) ⊔ ((b Δ c) \ a) : symm_diff_def _ _
+           ... = (a \ (b ⊔ c)) ⊔ (a ⊓ b ⊓ c) ⊔
+                   (b \ (c ⊔ a) ⊔ c \ (b ⊔ a))   : by rw [sdiff_symm_diff', symm_diff_sdiff]
+           ... = (a \ (b ⊔ c)) ⊔ (b \ (a ⊔ c)) ⊔
+                   (c \ (a ⊔ b)) ⊔ (a ⊓ b ⊓ c)   : by ac_refl
+
+lemma symm_diff_assoc : a Δ b Δ c = a Δ (b Δ c) :=
+by rw [symm_diff_symm_diff_left, symm_diff_symm_diff_right]
+
+end generalized_boolean_algebra
+
+section boolean_algebra
+variables {α : Type*} [boolean_algebra α] (a b c : α)
+
+lemma symm_diff_eq : a Δ b = (a ⊓ bᶜ) ⊔ (b ⊓ aᶜ) := by simp only [(Δ), sdiff_eq]
+
+lemma compl_symm_diff : (a Δ b)ᶜ = (a ⊓ b) ⊔ (aᶜ ⊓ bᶜ) :=
+by simp only [←top_sdiff, sdiff_symm_diff, top_inf_eq]
+
+lemma symm_diff_symm_diff_right' :
+  a Δ (b Δ c) = (a ⊓ b ⊓ c) ⊔ (a ⊓ bᶜ ⊓ cᶜ) ⊔ (aᶜ ⊓ b ⊓ cᶜ) ⊔ (aᶜ ⊓ bᶜ ⊓ c) :=
+calc a Δ (b Δ c) = (a ⊓ ((b ⊓ c) ⊔ (bᶜ ⊓ cᶜ))) ⊔
+                     (((b ⊓ cᶜ) ⊔ (c ⊓ bᶜ)) ⊓ aᶜ)  : by rw [symm_diff_eq, compl_symm_diff,
+                                                            symm_diff_eq]
+             ... = (a ⊓ b ⊓ c) ⊔ (a ⊓ bᶜ ⊓ cᶜ) ⊔
+                     (b ⊓ cᶜ ⊓ aᶜ) ⊔ (c ⊓ bᶜ ⊓ aᶜ) : by rw [inf_sup_left, inf_sup_right,
+                                                            ←sup_assoc, ←inf_assoc, ←inf_assoc]
+             ... = (a ⊓ b ⊓ c) ⊔ (a ⊓ bᶜ ⊓ cᶜ) ⊔
+                     (aᶜ ⊓ b ⊓ cᶜ) ⊔ (aᶜ ⊓ bᶜ ⊓ c) : begin
+                                                       congr' 1,
+                                                       { congr' 1,
+                                                         rw [inf_comm, inf_assoc], },
+                                                       { apply inf_left_right_swap }
+                                                     end
+
```

Comment:
I think I'm missing lemmas like `symm_diff_self`, `symm_diff_univ`, `symm_diff_empty`, `univ_symm_diff`, `empty_symm_diff`.
I think those would be great `@[simp]`-lemmas.

Reply:
Presumably it should be `top` and `bot` instead of `univ` and `empty`, but otherwise I agree those lemmas look like good additions.

Reply:
if you want, also `symm_diff_le_sup`, `disjoint_symm_diff_inf` `symm_diff_eq_left` (iff right is `bot`), `symm_diff_eq_right`, and injectivity in both arguments.

Reply:
Other than that, LGTM (my suggestion need not be added to this PR if you don't want, but I think @jcommelin's should.

Reply:
Thanks! I think I've added all the suggested lemmas; let me know if there are any others I should add.